src/execution/index/art/art.cpp

```diff
@@ -379,12 +382,14 @@ void GenerateKeysInternal(ArenaAllocator &allocator, DataChunk &input, unsafe_ve
 
 template <>
 void ART::GenerateKeys<>(ArenaAllocator &allocator, DataChunk &input, unsafe_vector<ARTKey> &keys) {
-	GenerateKeysInternal<false>(allocator, input, keys);
+	auto max_len = MAX_KEY_LEN * idx_t(prefix_count);
+	GenerateKeysInternal<false>(allocator, input, keys, max_len);
 }
 
 template <>
 void ART::GenerateKeys<true>(ArenaAllocator &allocator, DataChunk &input, unsafe_vector<ARTKey> &keys) {
-	GenerateKeysInternal<true>(allocator, input, keys);
+	auto max_len = MAX_KEY_LEN * idx_t(prefix_count);
```

Comment:
I don't really follow why we want to add the `prefix_count` here instead of having a general maximum value - could you expand on that? Wouldn't that cause issues if the prefixes are broken up when we e.g. insert new keys?

Reply:
I went for this solution because it scales with the `prefix_count` (which is fixed once the ART is created, as it is set by looking at the key types). So, for compound keys, or `VARCHAR` keys (16 bytes), we can have a higher `max_len`, because more bytes fit on a prefix node, so the stack is shorter.

Reply:
Alright, that makes sense